### PR TITLE
raw file reader

### PR DESCRIPTION
**Pull request by rmillikin**

### CODE REVIEW

**Comment by acesnik** [comment]
What's going on here?

**Comment by acesnik** [comment]
Oh, this is cool. Have you tried this out in MM locally?

**Comment by rmillikin** [comment]
there are some problems reading some ITMS data, I think it's a bug in rawfilereader but I haven't resolved it

**Comment by rmillikin** [comment]
I think we should merge this in soon and try to get all the NuGet stuff working and try it out in MetaMorpheus. The licence that we're required to bundle with this might not pack into NuGet...

RawFileReader has some issues centroiding one of the test datafiles but I don't know how common this is.

It's much faster at file loading than the old system and doesn't require the user to install anything.

**Comment by rmillikin** [comment]
We can use RawFileReader for opening most files, and if it crashes then we can use MSFileReader as a backup

File: ThermoRawFileReader/ThermoRawFileReaderData.cs [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        {
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException();
+            }
+
+            Loaders.LoadElements();
+
+            // I don't know why this line needs to be here, but it does...
+            var temp = RawFileReaderAdapter.FileFactory(filePath);
```
Comment (zrolfs) [inline ThermoRawFileReader/ThermoRawFileReaderData.cs]: weird

File: ThermoRawFileReader/ThermoRawFileReaderData.cs [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+            {
+                return null;
+            }
+
+            return GetOneBasedScan(dynamicConnection, filterParams, oneBasedScanNumber);
+        }
+
+        private static MsDataScan GetOneBasedScan(IRawDataPlus rawFile, IFilteringParams filteringParams, int scanNumber)
+        {
+            var scan = Scan.FromFile(rawFile, scanNumber);
```
Comment (zrolfs) [inline ThermoRawFileReader/ThermoRawFileReaderData.cs]: is this used for anything?

Comment (rmillikin) [reply ThermoRawFileReader/ThermoRawFileReaderData.cs]: I don't think so

**Comment by rmillikin** [comment]
don't merge this yet, I fixed the problem with reading ITMS files

**Comment by rmillikin** [comment]
I fixed reading ITMS data. just to be safe, I used MSConvert to convert the three .raw files to .mzML, read in that .mzML along with the .raw, and compared the peak count, intensities, and m/z values. All looks good

**Pull request merged**